ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java

```diff
@@ -116,66 +112,75 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
             blockParam.getGasLimit(),
             blockParam.getGasUsed(),
             blockParam.getTimestamp(),
-            Bytes.fromHexString(blockParam.getExtraData()),
+            Bytes.fromHexString(
+                blockParam.getExtraData()), // TODO: what should happen if extra data is null?
             blockParam.getBaseFeePerGas(),
             blockParam.getRandom(),
             0,
             headerFunctions);
 
-    String errorMessage = null;
-
     // ensure the block hash matches the blockParam hash
+    // this must be done before any other check
     if (!newBlockHeader.getHash().equals(blockParam.getBlockHash())) {
-      errorMessage =
+      LOG.debug(
           String.format(
               "Computed block hash %s does not match block hash parameter %s",
-              newBlockHeader.getBlockHash(), blockParam.getBlockHash());
+              newBlockHeader.getBlockHash(), blockParam.getBlockHash()));
+      return respondWith(reqId, null, INVALID_BLOCK_HASH);
     } else {
-      // do we already have this payload?
+      // do we already have this payload
       if (protocolContext
           .getBlockchain()
           .getBlockByHash(newBlockHeader.getBlockHash())
           .isPresent()) {
-        LOG.debug("block {} already present", newBlockHeader.getBlockHash());
-        return respondWith(reqId, blockParam.getBlockHash(), VALID, null);
+        LOG.debug("block already present");
+        return respondWith(reqId, blockParam.getBlockHash(), VALID);
       }
     }
 
-    final var latestValidAncestor = mergeCoordinator.getLatestValidAncestor(newBlockHeader);
-
-    if (latestValidAncestor.isEmpty()) {
-      return respondWith(reqId, null, SYNCING, null);
-    }
-
-    if (errorMessage != null) {
-      return respondWith(reqId, latestValidAncestor.get(), INVALID, errorMessage);
-    }
-
     // TODO: post-merge cleanup
     if (!mergeCoordinator.latestValidAncestorDescendsFromTerminal(newBlockHeader)) {
-      return new JsonRpcErrorResponse(
-          requestContext.getRequest().getId(), JsonRpcError.INVALID_TERMINAL_BLOCK);
+      return respondWith(requestContext.getRequest().getId(), null, INVALID_TERMINAL_BLOCK);
+    }
+
+    if (mergeContext.isSyncing()) {
```

Comment:
Unlike kintsugi, kiln spec says we should respond with SYNCING if we are syncing at all (whether backward or forward).  

Backward sync is tracked separately in mergeCoordinator, so we can handle that differently.  We should append the newPayload hash to the backward sync.  Something like:
```suggestion
    if (mergeContext.isSyncing() || mergeCoordinator.ifBackwardSyncing(newBlock)) {
      ...
```
 and in mergeCoordinator something like:
`public boolean ifBackwardsSyncing(Block blockToAppend) {}`

Reply:
The other way to implement the append behavior would be to let it flow into `executeBlock` like we did in kintsugi, and rather than returning a validation error, we let the append happen there and return a `Result` from `mergeCoordinator.executeBlock(..)` that indicates we are backward syncing

Reply:
I've for now added a stub method:
```
public boolean isBackwardSyncing(final Block blockToAppend) {
    return isBackwardSyncing();
  }
```

Do you think that is enough for now?